sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkTracer.java

```diff
@@ -33,6 +32,19 @@ final class SdkTracer implements ExtendedTracer {
     this.tracerEnabled = tracerConfig.isEnabled();
   }
 
+  static SdkTracer create(
+      TracerSharedState sharedState,
+      InstrumentationScopeInfo instrumentationScopeInfo,
+      TracerConfig tracerConfig) {
+    try {
+      Class.forName("io.opentelemetry.api.incubator.trace.ExtendedTracer");
+      return IncubatingUtil.createIncubatingTracer(
+          sharedState, instrumentationScopeInfo, tracerConfig);
+    } catch (Exception e) {
+      return new SdkTracer(sharedState, instrumentationScopeInfo, tracerConfig);
+    }
```

Comment:
maybe?

```suggestion
    if (IncubatingUtil.isPresent()) {
      return IncubatingUtil.createIncubatingTracer(
          sharedState, instrumentationScopeInfo, tracerConfig);
    } else {
      return new SdkTracer(sharedState, instrumentationScopeInfo, tracerConfig);
    }
```

Reply:
Yup yup that's my plan

Reply:
Tried this and a bunch of different variations of it. 

The organization of methods is pretty finicky to get this to work, and it appears that by placing the `isPresent()` check in the same class file as the `createIncubatingTracer` method, the JVM tries to load the incubating classes required by `createIncubatingTracer` even if `isPresent()` is false.

Based on the stack trace, I think what is happening is the JVM reaches `isPresent()`, and tries to load _all classes_ required by all methods in `IncubatingUtil`. 

This problem occurs even if I use fully qualified class named in `createINcubatingTracer`, rather than imports.

---

Another thing to think about is how graal interacts with this increased use of reflection. I believe we'll have to add new [reflect-config.json](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/incubator/src/main/resources/META-INF/native-image/io.opentelemetry/opentelemetry-api/reflect-config.json) files to inform the graal runtime.